lib/src/ui/pages/authentication/registration/registration_notifier.dart

```diff
@@ -0,0 +1,93 @@
+import 'package:allthenews/generated/l10n.dart';
+import 'package:allthenews/src/domain/authorization/authentication_repository.dart';
+import 'package:allthenews/src/domain/communication/exception_mapper.dart';
+import 'package:allthenews/src/ui/pages/authentication/registration/registration_state.dart';
+import 'package:flutter/cupertino.dart';
+
+class RegistrationNotifier extends ChangeNotifier {
+  final AuthenticationRepository _authorizationRepository;
+  final ExceptionMapper _exceptionMapper;
+
+  RegistrationNotifier(this._authorizationRepository, this._exceptionMapper);
+
+  RegistrationState _state = const RegistrationState();
+
+  RegistrationState get state => _state;
+
+  VoidCallback returnToProfile;
+
+  Future<void> createUser(String email, String password, String name) async {
+    _validateFields();
```

Comment:
Średnio mi się podoba bo ta walidacja częściowo jest w tej metodzie a częściowo w tym kolejnym ifie. Może jakby sygnatura wyglądała mniej więcej tak to zwięźlej by to wyglądało? 🤔 Taki luźny pomysł
```
_validateFields(
   onInvalid: (validationErrors) => ...,
   onValid: (state) => ...
)
```